src/features/game/actions/onChainEvents.ts

```diff
@@ -0,0 +1,177 @@
+import { fromWei } from "web3-utils";
+
+import { metamask } from "lib/blockchain/metamask";
+import { shortAddress } from "lib/utils/shortAddress";
+
+import token from "assets/icons/token.gif";
+
+import { ITEM_DETAILS } from "../types/images";
+import { KNOWN_ITEMS } from "../types";
+import { getItemUnit } from "../lib/conversion";
+
+export type OnChainEvent = {
+  icon: string;
+  message: string;
+  blockNumber: number;
+  transactionHash: string;
+};
+
+type LoadPastEventArgs = {
+  farmId: number;
+  farmAddress: string;
+};
+
+/**
+ * Load On Chain events and transform them into OnChainEvent[]
+ */
+async function loadPastEvents({ farmId, farmAddress }: LoadPastEventArgs) {
```

Comment:
Does it make sense to always grab all past events. If we store the block number should we just grab events from the last block number?